lib/gollum/public/gollum/javascript/editor/langs/rest.js

```diff
@@ -7,9 +7,15 @@
 
 
   var headerFunc = function (selText, selectedRange, symbol) {
-    adornment = symbol.repeat(selText.length);
-    repText = selText + "\n" + adornment + "\n";
-    $.GollumEditor.replaceSelection(repText, false, false, selectedRange);
+      var num = 10;
```

Comment:
Why is this value hardcoded? What does it signify exactly?

Reply:
I noticed if you clicked a rst header button without having any text on a line, nothing would happen. In other markups, this would just give you the header syntax without any text. This is because the rst header function produces a number of header-characters (e.g. `=`) based on the number of characters in the title (rst requires at least as many header-characters as the title length). So if you didn't have a title yet, the buttons generated 0 adornment characters. But that also means you couldn't use the header buttons to 'find out' about proper syntax in rst. `10` is just an arbitrary default that. We could also stick with the original behavior of generating no header syntax when the title is empty.

Reply:
I see. What about generating a dummy title if there is no text yet?

`Title here`
`==========`

Reply:
The crucial part of the syntax is of course that the adornments have to be at least as long as the title. This comes out a little better with a dummy title than with an arbitrary number of 10 I think.

Reply:
:+1: